.github/workflows/deploy.yaml

```
@@ -55,9 +55,9 @@ jobs:
           yq e -i ".image.tag = \"${{ inputs.version }}\"" values/${{ inputs.values_folder }}/${{ inputs.service }}.yaml
       - name: Update app metadata
         run: |
-          branch=${{ inputs.branch }}
-          if [[ ${{ inputs.new_branch }} != '' ]]; then
-            branch=${{ inputs.new_branch }}
+          branch=${{ inputs.new_branch }}
+          if [[ $branch == '' ]]; then
+            branch=${{ inputs.branch }}
```

I'd still wrap all the variables in double quotes for being paranoid, especially after the last fiasco. But that's up to you, of course :) 
```suggestion
          branch="${{ inputs.new_branch }}"
          if [[ "$branch" ]]; then
            branch="${{ inputs.branch }}"
```